File: app/src/main/java/com/github/kongpf8848/animation/adapter/ViewPagerAdapter.java
package com.github.kongpf8848.animation.adapter;

import androidx.viewpager.widget.PagerAdapter;
import androidx.viewpager.widget.ViewPager;
import android.view.View;

import java.util.List;

/**
 * Created by pengf on 2016/12/15.
 */

public class ViewPagerAdapter extends PagerAdapter {
    public List<View> viewList;

    public ViewPagerAdapter(List<View> viewList) {
        this.viewList = viewList;
    }

    @Override
    public int getCount() {
        if (viewList != null) {
            return viewList.size();
        }
        return 0;
    }

    @Override
    public boolean isViewFromObject(View view, Object object) {
        return view == object;
    }

    @Override
    public Object instantiateItem(View container, int position) {

        if (this.viewList.get(position).getParent() == null)
        {
            ((ViewPager)container).addView(this.viewList.get(position), 0);
        }
        return this.viewList.get(position);

    }
    @Override
    public void destroyItem(View container, int position, Object object)
    {
        ((ViewPager)container).removeView(this.viewList.get(position));
    }
}

File: app/src/main/java/com/github/kongpf8848/animation/adapter/ViewPagerAdapter.kt
package com.github.kongpf8848.animation.adapter

import android.util.Log
import android.view.View
import android.view.ViewGroup
import androidx.viewpager.widget.PagerAdapter
import androidx.viewpager.widget.ViewPager

class ViewPagerAdapter(val viewList: List<View>) : PagerAdapter() {

    private val TAG = "ViewPagerAdapter"

    override fun getCount(): Int {
        Log.d(TAG, "getCount() called")
        return viewList.size
    }

    override fun isViewFromObject(view: View, `object`: Any): Boolean {
        Log.d(TAG, "isViewFromObject() called with: view = $view, object = $`object`")
        return view === `object`
    }

    override fun instantiateItem(container: ViewGroup, position: Int): Any {
        Log.d(TAG, "instantiateItem() called with: container = $container, position = $position")
        container.addView(viewList[position])
        return viewList[position]
    }


    override fun destroyItem(container: ViewGroup, position: Int, `object`: Any) {
        Log.d(TAG, "destroyItem() called with: container = $container, position = $position, object = $`object`")
        container.removeView(`object` as View)
    }

}